File: 0101-0200/0171.Excel_Sheet_Column_Number.py
#0ms





class Solution:
    def titleToNumber(self, columnTitle: str) -> int:
        ans, pos = 0, 0
        for letter in reversed(columnTitle):
            digit = ord(letter)-64
            ans += digit * 26**pos
            pos += 1
            
        return ans






#0ms







class Solution:
    def titleToNumber(self, columnTitle: str) -> int:
        col_num = 0
        for char in columnTitle:
            col_num = col_num * 26 + (ord(char) - 64)
        return col_num





#0ms






class Solution:
    def titleToNumber(self, columnTitle: str) -> int:
        result=0
        for char in columnTitle:
            result=result*26+(ord(char)-ord('A')+1)
        return result

File: 0101-0200/0171.Excel_Sheet_Column_Number.rs
//0ms





impl Solution {
    pub fn title_to_number(column_title: String) -> i32 {
        column_title.bytes().fold(0, |acc, b| acc * 26 + (b - 64) as i32)
    }
}





//0ms






fn alpha_to_num(c: char) -> i32 {
    (c as i32) - ('A' as i32) + 1
}
impl Solution {
    pub fn title_to_number(column_title: String) -> i32 {
        let mut acc: i32 = 0;
        for c in column_title.chars() {
            acc = 26 * acc + alpha_to_num(c);
        }

        return acc;
    }
}
